### ISSUE TITLE
Regularly disappears from status bar+becomes unresponsive (MacOS High Sierra)

### ISSUE THREAD

**Issue by mattwkelly** [sfsam/Itsycal#79]
I'm on MacOS High Sierra 10.13. When clicked, itsycal becomes unresponsive for ~10 seconds (spinning beach ball) and then finally appears. Also, it regularly disappears from the status, even though it appears in activity monitor. Killing the app and relaunching fixes it (for a while).

**Comment by sfsam** [comment]
Does it use an excessive amount of memory? I'm wondering if this bug is a duplicate of https://github.com/sfsam/Itsycal/issues/54.

**Comment by mattwkelly** [comment]
Yes, it appears that's the case. Here's what it looks like after having it open for ~10 seconds and clicking on it a dozen or so times.
<img width="590" alt="screen shot 2017-10-19 at 11 37 16 am" src="https://user-images.githubusercontent.com/174449/31787914-fba5cf02-b4c1-11e7-98cb-985414d7a4c2.png">

**Comment by sfsam** [comment]
Could you please try the following:

1. Quit Itsycal.
2. Go to System Preferences... > Security & Privacy > Privacy > Calendars and uncheck Itsycal. This revokes calendar access to Itsycal so it can no longer display your events.
3. Restart Itsycal and observe memory usage for a while. What happens?
4. Quit Itsycal.
5. Go back to System Preferences and regrant calendar access.
6. Restart Itsycal and observe memory usage for a while. What happens?

**Comment by mattwkelly** [comment]
#3: Memory usage stays around ~15mb.
#6: Memory usage keeps increasing.

So it appears it's the Mac Calendar integration causing issues.

**Comment by Sturek** [comment]
I have the same issue. The icon does not disappear from my menubar, but other than that, the same issues. As I have an older MacBook Air, the memory consumption makes itsycal unusuable pretty quickly, unfortunately.

**Comment by sfsam** [comment]
I've put together a test version of Itsycal that might address the memory leak. I still have no idea what is causing it, so this is a shot in the dark. I've had one user test it so far with promising results and another with not-so promising results.

https://s3.amazonaws.com/itsycal/Itsycal-test.zip

Please let me know if it makes a difference.

**Comment by adamweeks** [comment]
High Sierra 10.13.1 seems to have resolved the issue for me. Memory usage has held at under 50MB while running the past week.

**Comment by Sturek** [comment]
This seems to have solved it for me as well. Still using the regular version of itsycal.

**Comment by mattwkelly** [comment]
I'm still seeing the issue on High Sierra 10.13.1, it just takes a couple of hours instead of minutes to balloon memory usage. (At the time of this comment, it's at 2.59GB after ~2 hours)

**Comment by sfsam** [comment]
I don't know the cause, but I don't think 10.13.1 has anything to do with it. I think it's a coincidence. It seems to be something transient where it affects some people for a while and then goes away. I haven't seen it yet on my end and I'm still stumped. Did you try the test version?

**Comment by Sturek** [comment]
You were right, after one more day, the memory usage was too high again. I've been running the test version you posted for two days now and am still at 29 MB. I'll continue to test, but it's looking good so far, the number changes far less already.

**Comment by mattwkelly** [comment]
I'm trying the test version now and will follow up. Thanks!

**Comment by Sturek** [comment]
Over the weekend, the test version has now racked up more than 100 megs. It's still usable, but becoming slower. Sorry.

**Comment by sfsam** [comment]
@Sturek et al: Thanks for checking. I updated the comment above to reflect that the test version does not appear to resolve this issue.

**Comment by mattwkelly** [comment]
I started the test version ~12 hours ago and it's at 586MB. So it's steadily increasing as well, although at a much slower pace.

**Comment by tomharrisonjr** [comment]
I ran into this and presumably #54 as well today.  I have disabled calendar access in Privacy settings and memory seems to be steady.  macOS 10.13.2 with all patches as of 2017-12-18

Is there any chance that the specific calendar services behind this issue?  I have had many, many issues over the last few years with the integration of our company's Office 365 calendar (Exchange/Outlook) with macOS calendar.  I also sync with my personal Google calendar, plus a couple of shared calendars.  Just providing some details in case this is helpful in isolating the issue.
